Distribution/Server/Features/Upload.hs

```diff
@@ -298,6 +301,18 @@ uploadFeature ServerEnv{serverBlobStore = store}
     guardAuthorisedAsMaintainer pkgname =
       guardAuthorised_ [InGroup (maintainersGroup pkgname)]
 
+    guardAuthorisedAsTrustee :: ServerPartE ()
+    guardAuthorisedAsTrustee =
+        guardAuthorised_ [InGroup trusteesGroup]
+
+    authorisedAsMaintainerOrTrustee :: PackageName -> ServerPartE Bool
+    authorisedAsMaintainerOrTrustee pkgname=
+        guardAuthorised' [InGroup (maintainersGroup pkgname), InGroup trusteesGroup]
+
+    authorisedAsAnyUser :: ServerPartE Bool
+    authorisedAsAnyUser =
+        guardAuthorised' [AnyKnownUser]
```

Comment:
I don't want to add these new auth functions, as mentioned above, but adding a wrapper here for `checkPriviledged` here is ok. e.g.
```
checkPriviledged :: UserId -> [PrivilegeCondition] -> ServerPartE Bool
checkPriviledged uid privconds = do
    users <- queryGetUserDb
    Auth.checkPriviledged users uid privconds
```